lib/toSyllabicSymbols.js

```diff
@@ -0,0 +1,127 @@
+const toUnicode = require( './toUnicode' )
+const { getRegexGroup } = require( './regex-utils' )
+
+// Remove 0 weight syllables from the input string before counting
+const syllabicStripping = [
+  '੍ਰ',
+  'ੵ',
+  'ੑ',
+  '੍ਵ',
+  '੍ਹ',
+  '੍ਟ',
+  '੍ਨ',
+  '੍ਯ',
+  '੍ਚ',
+  '੍ਤ',
+  '੍',
+  'ਿ',
+  'ੁ',
+  'ਂ',
+  'ਃ',
+  '।',
+  '॥',
+  '☬',
+  '਼',
+  '❁',
+]
+
+const syllabicMapping = {
+  ਇ: 1,
```

Comment:
Instead of repeatedly using string value results, reference these by defining them as an object and referencing the value.

A way to make this more succinct would be to leverage the object you made, and use it as such:

```
const syllabicSymbols = {
  short: 1
  long : 2
  something: a,
  else: b,
}

// Instead of redeclaring { letter: syllabicSymvol.something } for every letter, just group them into arrays and expand them with a reducer
const syllabicMapping = Object.entries( {
  [syllabicSymbols.short]: [ 'ਉ', 'ਊ' ],
// etc
} ).reduce( ( symbols, [ syllabicSymbol, letter ] ) => ( {
  ...symbols,
  [ letter ]: syllabicSymbol,
}, {} )

```

Reply:
I can do this for readability

Reply:
Please let me know if it's better now